src/SketchField.jsx

```diff
@@ -282,6 +280,28 @@ class SketchField extends PureComponent {
         } else if (canvas.getHeight() < this.props.fixCanvasToSize.height) {
             canvas.setHeight(this.props.fixCanvasToSize.height);
         }
+
+        let xTransform = canvas.viewportTransform[4] + canvas.getWidth() - prevWidth;
```

Comment:
Accessing an array via index could be sketchy. Consider checking the length of canvas.viewportTransform to ensure there is something there.